### PR TITLE
xcodes: build from source

### PR DESCRIPTION
**Pull request by emilytrau**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/darwin/apple-sdk-11.0/libcompression.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{ stdenvNoCC, buildPackages, MacOSX-SDK }:
+
+let self = stdenvNoCC.mkDerivation {
+  pname = "libcompression";
```
Comment (stephank) [inline pkgs/os-specific/darwin/apple-sdk-11.0/libcompression.nix:4]: Oh, I think this is probably a better solution than https://github.com/NixOS/nixpkgs/pull/230390/files#diff-3e4b9f931538d331c1402357df553700a107e167d0a6197a53e6c85e3e42aa03R252

**Review by stephank — COMMENTED** [review]
I get:

```
error: hash mismatch in fixed-output derivation '/nix/store/ing2sqh1gb6l17jznyf0gcdyzh9pwy85-PromiseKit-1c296a8.drv':
         specified: sha256-iJvejh0bsYEZlixllcbNWjYti8Rh1lSJGF1xZr39ZgU=
            got:    sha256-zRxA++Pz/d79i023duA5047wsCqxUaExaevJqhxpk24=
```

**Review by philiptaron — CHANGES_REQUESTED** [review]
Like @stephank, I get a build error:

```
$ nix build --nix-path nixpkgs=/Users/philip/.cache/nixpkgs-review/pr-245205/nixpkgs nixpkgs-overlays=/var/folders/2b/x9fgc9_s5bg24d7wm_5txz180000gn/T/tmpoefsy9ab --extra-experimental-features nix-command no-url-literals --no-link --keep-going --no-allow-import-from-derivation -f /Users/philip/.cache/nixpkgs-review/pr-245205/build.nix
error: hash mismatch in fixed-output derivation '/nix/store/hxb2rv5rcfydw515hpyvmjmcr73j1pfw-PromiseKit-1c296a8.drv':
         specified: sha256-iJvejh0bsYEZlixllcbNWjYti8Rh1lSJGF1xZr39ZgU=
            got:    sha256-zRxA++Pz/d79i023duA5047wsCqxUaExaevJqhxpk24=
error: 1 dependencies of derivation '/nix/store/5207dcpbwck245hr8rgnf4iqllm7zv16-xcodes-1.4.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vq0yj72z1xw7hl265h1w5wkzf2kpxlag-review-shell.drv' failed to build

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/245205

1 package failed to build:
xcodes
```

I like this PR, though.

**Comment by emilytrau** [comment]
Thanks for reviewing! I tracked down the hash mismatch to a bug in swiftpm2nix. Hopefully it should be reproducible for you now?

**Review by philiptaron — APPROVED** [review]
Thanks for the diagnosis and bug fix!

Result of `nixpkgs-review pr 245205` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>swiftpm2nix</li>
    <li>xcodes</li>
  </ul>

**Pull request merged**